### PR TITLE
GH-34869: [C++] Configure alpine linux nightly job to build gtest from source

### LINKED ISSUE #34869: [C++] Alpine linux nightly job is failing due to an invalid gtest
### Describe the bug, including details regarding any error messages, version, and platform.

It appears that the gtest that ships with Alpine does not meet the C++17 check.  We can just download it.

### Component(s)

C++

### PR DESCRIPTION
**Pull request by westonpace**
### Rationale for this change

The nightly test is currently failing with an error:

```
-- Found GTest: /usr/lib/cmake/GTest/GTestConfig.cmake (found suitable version "1.11.0", minimum required is "1.10.0")  
CMake Error at cmake_modules/ThirdpartyToolchain.cmake:2236 (message):
  System GTest is built with a C++ standard lower than 17.  Use bundled GTest
  via passing in CMake flag

  -DGTest_SOURCE="BUNDLED"
```

### What changes are included in this PR?

Changes the alpine linux nightly test to use `-DGTest_SOURCE="BUNDLED"`

### Are these changes tested?

Yes, by the nightly test itself.

### Are there any user-facing changes?

No
* Closes: #34869

### CODE REVIEW

**Comment by westonpace** [comment]
@github-actions crossbow submit test-alpine-linux-cpp

File: docker-compose.yml [hunk 1]
```diff
@@ -223,6 +223,8 @@ services:
       <<: *ccache
       ARROW_ENABLE_TIMING_TESTS:  # inherit
       ARROW_MIMALLOC: "ON"
+      GTest_SOURCE: "BUNDLED"  # Alpine's GTest is not built against
+                               # C++17 so we build from source
```
Comment (kou) [inline docker-compose.yml:227]: Could you move this to `ci/docker/alpine-linux-3.16-cpp.dockerfile` like other `*_SOURCE=BUNDLED`?

Comment (westonpace) [reply docker-compose.yml]: Good catch.  I have done this.

**Comment by westonpace** [comment]
@github-actions crossbow submit test-alpine-linux-cpp

**Review by kou — APPROVED** [review]
+1

**Pull request merged**